### ISSUE TITLE
HTML link in JSON

### ISSUE THREAD

**Issue by oliveskur** [notepad-plus-plus/notepad-plus-plus#13522]
### Description of the Issue

If link ends with escaped quote then remaining part of line is marked as invalid syntax but actually it passes any JSON parser.

### Steps to Reproduce the Issue

1.  Create a new json file
2. Paste following: {"html":"<a href=\\"https://www.google.com\" target=\\"_blank\\">Google</a>"}

### Expected Behavior

It's correct syntax

### Actual Behavior

target=\"_blank\">Google</a>"} is marked as invalid syntax (red background)

### Debug Information

Notepad++ v8.5.2   (64-bit)
Build time : Apr  4 2023 - 19:55:32
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "C:\Users\olive\OneDrive\Desktop\test.json"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Pro (64-bit)
OS Version : 22H2
OS Build : 19045.2846
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)

**Comment by rdipardo** [comment]
Duplicate of #11746, #11467 and ScintillaOrg/lexilla#72.

**Comment by dail8859** [comment]
It looks like the appropriate json lexer property just needs set if I understand this correctly. Example LuaScript command:

```lua
editor.Property["lexer.json.escape.sequence"] = "1"
```
![json](https://user-images.githubusercontent.com/3694843/232590505-683ef5a1-8e14-4281-8e17-37ca0bd56d4e.gif)

**Comment by dail8859** [comment]
@rdipardo Oh I see you posted in the Lexilla issue regarding this exact property. Guess the issue is deeper than I realized.

**Comment by rdipardo** [comment]
> Guess the issue is deeper than I realized.

IIUC, JSON strings are supposed to terminate *even when* the closing quote is escaped.
In other words, `\"` can be a *compound delimiter*, not just a string continuation sequence as in most C-like languages.   

~~~sh-session
$ node
Welcome to Node.js v18.13.0.
Type ".help" for more information.

$ let x = {"html":"<a href=\"https://www.google.com\" target=\"_blank\">Google"}
undefined

$ x.html
'<a href="https://www.google.com" target="_blank">Google'
~~~

**Comment by byronigoe** [comment]
Is LuaScript the only way to set `lexer.json.escape.sequence=1`? I would like to make this change in the config.xml and it be persistent.

**Comment by dail8859** [comment]
@byronigoe It requires a plugin, there's several that can do it but all require custom functionality to make it work. There may be some easier to use plugin for setting it but I'm not aware of any currently. There's no options exposed in the config.xml file to set this option.

**Comment by byronigoe** [comment]
Thanks for the quick response. Is there anything I can do to help make progress toward a proper resolution? Does this need to be resolved in Lexilla? Should an option be added to NPP?

**Comment by dail8859** [comment]
At least to get the functionality I demonstrated with LuaScript that would need to be added to Npp. Other languages in Npp have properties set in the code so it should be fairly straightforward.

As @rdipardo pointed out there is a deeper issue here that would need addressed by Lexilla. Since I don't fully understand `lexer.json.escape.sequence` I'm assuming it doesn't bring any negative side affects and is at least a step in the right direction.

**Comment by byronigoe** [comment]
Indeed, it looks like a simple change in the neighborhood of https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/lexilla/lexers/LexJSON.cxx#L122

**Comment by dail8859** [comment]
@byronigoe That's what defines what properties the JSON lexer implements. The exact code change would be required here, there's already a few other properties being set:

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/c76f178534bba518f263c8120b732801e13c4916/PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp#L724-L728
